src/main/java/slimeknights/tconstruct/gadgets/entity/shuriken/ShurikenEntityBase.java

```diff
@@ -0,0 +1,103 @@
+package slimeknights.tconstruct.gadgets.entity.shuriken;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.ProjectileItemEntity;
+import net.minecraft.entity.projectile.SnowballEntity;
+import net.minecraft.item.Item;
+import net.minecraft.network.IPacket;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.math.EntityRayTraceResult;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.registry.IEntityAdditionalSpawnData;
+import net.minecraftforge.fml.network.NetworkHooks;
+import slimeknights.tconstruct.gadgets.TinkerGadgets;
+
+import javax.annotation.Nonnull;
+
+public abstract class ShurikenEntityBase extends ProjectileItemEntity implements IEntityAdditionalSpawnData {
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, World worldIn) {
+    super(type, worldIn);
+  }
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, double x, double y, double z, World worldIn) {
+    super(type, x, y, z, worldIn);
+  }
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, LivingEntity livingEntityIn, World worldIn) {
+    super(type, livingEntityIn, worldIn);
+  }
+
+    /**
+   * Get damage dealt by Shuriken
+   * Should be <= 20.0F
+   * @return float damage
+   */
+  public abstract float getDamage();
+
+  /**
+   * Get knockback dealt by Shuriken
+   * Should be <= 1.0F, Minecraft
+   * typically uses values from 0.2F-0.6F
+   * @return float knockback
+   */
+  public abstract float getKnockback();
+
+  /**
+   * Get the entity, flint or quartz,
+   * to use in onImpact method.
+   * @return ShurikenEntityBase entity
+   */
+  public abstract ShurikenEntityBase getShurikenEntity();
+
+  @Override
+  protected void onImpact(RayTraceResult result) {
+    super.onImpact(result);
+
+    if (!this.world.isRemote) {
+      if (result.getType() == RayTraceResult.Type.BLOCK) {
```

Comment:
Handled in super class. Code is as follows:

```java
      RayTraceResult.Type raytraceresult$type = result.getType();
      if (raytraceresult$type == RayTraceResult.Type.ENTITY) {
         this.onEntityHit((EntityRayTraceResult)result);
      } else if (raytraceresult$type == RayTraceResult.Type.BLOCK) {
         this.func_230299_a_((BlockRayTraceResult)result);
      }
```

So looks like `func_230299_a_` is the method you want to override for the block hit.